### PR TITLE
Feat: New grub2_editenv_list parser

### PR DESCRIPTION
**Pull request by TZ3070**
Signed-off-by: Xinting Li <xintli@redhat.com>

### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?

### Complete Description of Additions/Changes:

New parser for command 'grub2-editenv list'

### CODE REVIEW

**Comment by xiangce** [comment]
@TZ3070 - Is this ready for review?

For new spec, please raise an MR/issue to the assets repo first to get it approved at first. Thanks.

File: insights/parsers/grub2_editenv_list.py
Comment (ryan-blakley) [inline insights/parsers/grub2_editenv_list.py:13]: @TZ3070 Since this is importing from the grubenv parser file, could you move this parser to that file instead of having a separate file just for this parser since they're both about grubenv.

Comment (TZ3070) [reply insights/parsers/grub2_editenv_list.py]: Updated

File: insights/parsers/grubenv.py
Comment (xiangce) [inline insights/parsers/grubenv.py:13]: For sub-title, `---` should be used.

File: insights/parsers/grubenv.py
Comment (xiangce) [inline insights/parsers/grubenv.py:132]: It seems "errors" is the only difference between the two parsers, it would be better to add a new base class and make them two inherit from the base.

**Review by xiangce — CHANGES_REQUESTED** [review]

File: insights/parsers/grubenv.py
Comment (TZ3070) [reply insights/parsers/grubenv.py]: @xiangce Updated

File: insights/parsers/grubenv.py
Comment (ryan-blakley) [inline insights/parsers/grubenv.py:44]: Not a big deal but since this is a list, could this be changed to self._errors so it's plural.

File: insights/parsers/grubenv.py
Comment (ryan-blakley) [inline insights/parsers/grubenv.py:57]: Could the check on errors be in a sub if statement, that way if there is no data but there were errors it should raise a ParseException, otherwise throw a SkipException. The way it is now, if there is no data parsed but an error line was stored it would return an empty dict when it shouldn't.

File: insights/parsers/grubenv.py
Comment (ryan-blakley) [inline insights/parsers/grubenv.py:146]: Could this be moved to the base class so both classes can use the property? Also could this be changed to errors since it's returning a list, and add an entry to the attributes docstring in the base class.

File: insights/parsers/grubenv.py
Comment (TZ3070) [reply insights/parsers/grubenv.py:57]: Thanks for your comment. 
For 'GrubEnv' parser, it processes the content in '/boot/grub2/grubenv' file. The file saves environment variables for boot time, and will never contain error messages.
For 'Grub2EditenvList' parser, it parses the output of the 'grub2-editenv list' command.  The list the current variables in environment block file if there are no errors in the file configuration. Otherwise, there will be no data and only error messages in output. If there is no data parsed but an error line was stored, the return is not only an empty dict, but also self.error which is not empty. And error message is one the information we needed.
```
#e.g.
[root@rhel~]# grub2-editenv list
grub2-editenv: error: invalid environment block.
```

From my point of view, if it is just no data parsed, it is not necessary to throw an exception. If there is no data and no error, the parser must throw an exception.

File: insights/parsers/grubenv.py
Comment (TZ3070) [reply insights/parsers/grubenv.py:146]: Like the comment I left above, the content of 'GrubEnv' parser will never contain error messages. So, the error attribute is no need to be added in base class for each parser to use.

File: insights/parsers/grubenv.py
Comment (TZ3070) [reply insights/parsers/grubenv.py]: @ryan-blakley Updated

File: insights/parsers/grubenv.py
Comment (xiangce) [reply insights/parsers/grubenv.py]: BTW, @TZ3070 - Because this is special processing, please add the above explanation to the code as a comment, to prevent someone from updating the parser in the future without understanding the logic.

Comment (xiangce) [reply insights/parsers/grubenv.py]: Oh, my fault. You have already added the comment.

**Review by xiangce — APPROVED** [review]
Approved pending the approval of spec

**Comment by xiangce** [comment]
@TZ3070 - since the existing `GrubEnv` and the new one is a bit relevant, would you please make a combiner for them based on the current implementation?

File: insights/parsers/grubenv.py
Comment (xiangce) [inline insights/parsers/grubenv.py:55]: The example section does not need a double colon

File: insights/specs/insights_archive.py
Comment (xiangce) [inline insights/specs/insights_archive.py:83]: Since the spec is changed to command, here the spec should be updated as:
```
grubenv = first_file(["insights_commands/grub2-editenv_list", "/boot/grub2/grubenv", "/boot/efi/EFI/redhat/grubenv"])
```

File: insights/tests/parsers/test_grubenv.py
Comment (xiangce) [inline insights/tests/parsers/test_grubenv.py:31]: I believe these kind of lines should be removed after the spec changed to command.

File: insights/parsers/grubenv.py
Comment (xiangce) [inline insights/parsers/grubenv.py:44]: It should be a list.

File: insights/parsers/grubenv.py
Comment (xiangce) [inline insights/parsers/grubenv.py:55]: The example section does not need a double colon

**Review by xiangce — CHANGES_REQUESTED** [review]

**Review by xiangce — APPROVED** [review]

**Pull request merged**
